services/movie.py

```diff
@@ -0,0 +1,47 @@
+from typing import Optional
+
+from django.db.models import QuerySet
+
+from db.models import Movie
+
+
+def get_movies(
+        genres_ids: Optional[list[int]] = None,
+        actors_ids: Optional[list[int]] = None
+) -> QuerySet:
+    query_set = Movie.objects.all()
+    if genres_ids and actors_ids:
+        return query_set.filter(
+            genres__id__in=genres_ids,
+            actors__id__in=actors_ids
+        )
+    if genres_ids:
+        return query_set.filter(
+            genres__id__in=genres_ids
+        )
+    if actors_ids:
+        return query_set.filter(
+            actors__id__in=actors_ids
+        )
+    return query_set

```

Comment:
let's filter queryset instead of making return. in this case each if statement overrides the initial queryset with some filter and then you don't need the first condition